data/transactions/logic/opcodes.go

```diff
@@ -61,38 +63,112 @@ const createdResourcesVersion = 6
 // experimental-
 const fidoVersion = 7 // base64, json, secp256r1
 
+type linearCost struct {
+	baseCost  int
+	chunkCost int
+	chunkSize int
+}
+
+func (lc *linearCost) compute(stack []stackValue) int {
```

Comment:
@jannotti Optional:  What are your thoughts on testing for % of maximum budget use by linear cost opcodes?

* The question is prompted by reviewing the PR's tests and a prior verbal around _not_ enabling linear costs for hashing opcodes (e.g. `sha256`).
* The intent is to flag (via failing test) that an opcode's max cost consumes an _unreasonable_ % of opcode budget.  I'm imagining later opcode changes where we don't fully consider change implications.
* I'm on the fence for at least a few reasons:
  * I can imagine circumstances where a failing test doesn't change our opinion.  And then the test becomes a nuisance.
  * To support writing a test that generically exercises all opcodes, the opcode model must have a max input length specified.  Adding such information might be considered extraneous.

Reply:
It's a nice idea. I would advocate only doing it for linear opcodes (so expensive opcodes like verify aren't even an issue). And if the test gives larger and larger inputs, opcodes like `b+` will fail under their own limits (which would be deemed a test success) so we would not have to encode the max limits.
A simpler thing would be to make
```
func costByLength(initial int, perChunk int, chunkSize int) opDetails {
	if initial < 1 || perChunk <= 0 || chunkSize < 1 || chunkSize > maxStringSize {
		panic("bad cost configuration")
	}
	return opDetails{linearCost{initial, perChunk, chunkSize}, 1, nil, nil, nil}
}
```
bail when called with parameters that lead to crazy costs.  That's currently the only way linearCosts are built with non-zero chunk parameters.

Reply:
@jannotti The thinking makes sense.  I think your suggestion to use `b+` to bound the test is the most expedient + robust route.

Resolving because the idea isn't needed for the PR.  I will keep the idea in mind.